torch_frame/data/mapper.py

```diff
@@ -129,7 +130,19 @@ def forward(
         if ser.dtype != 'object':
             raise ValueError('Multi-categorical types expect string as input')
         values = []
+        original_index = ser.index
         ser = ser.apply(self._row_to_tensor)
+        ser = ser.explode()
+        ser = pd.merge(
+            ser.rename('data'),
+            self.index.rename('index'),
+            how='left',
+            left_on='data',
+            right_index=True,
+        ).dropna()
+        ser['index'] = ser['index'].astype('int64')
+        ser = ser.groupby(level=0)['index'].apply(list)
```

Comment:
Why do we need to group here again? We already have the flattened output, right?

Reply:
Removed `groupby`